### DISCUSSION TITLE
ESP32 mDNS support on Ethernet (class LAN)

### DISCUSSION THREAD

**Discussion by mohnazemi** [micropython/micropython#12254] [ESP32] [answered]
Hi guys,

I wanted to know if there is a specific reason why the mDNS is supported on the WLAN interface (via hostname config parameter) for ESP32 but it is not supported on the LAN/Ethernet interface. There seems to be no limitation in IDF's TCP/IP Adapter library for this, but the ESP32 port does not expose a config parameter to set this. By the look of it, mDNS can be enabled without much work in `network_lan.c` similar to the way it is implemented in `network_wlan.c`

**Comment by jimmo** [comment, marked as answer]
@mohnazemi No specific reason, please send a PR if you can get this working!
